src/main/java/org/fenixedu/academic/service/services/enrollment/shift/WriteStudentAttendingCourse.java

```diff
@@ -40,7 +42,16 @@ protected void run(Registration registration, String executionCourseId) throws F
         if (registration == null) {
             throw new FenixServiceException("error.registration.not.exist");
         }
-        registration.addAttendsTo(readExecutionCourse(executionCourseId));
+
+        ExecutionCourse executionCourse = readExecutionCourse(executionCourseId);
+
+        for (Registration registration2 : registration.getStudent().getRegistrationsSet()){
```

Comment:
The getRegistrationsSet() method has the same problem of the getAllRegistrations() method.
Using getRegistrationStream() would be a better choice.